datasource/hcp-packer-version/data.go

```diff
@@ -0,0 +1,149 @@
+	err := config.Decode(&d.config, nil, raws...)
+	if err != nil {
+		return err
+	}
+
+	var errs *packersdk.MultiError
+
+	if d.config.BucketName == "" {
+		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("the `bucket_name` must be specified"))
+	}
+	if d.config.ChannelName == "" {
+		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("the `channel_name` must be specified"))
+	}
+
+	if errs != nil && len(errs.Errors) > 0 {
+		return errs
+	}
+	return nil
+}
+
+// DatasourceOutput is essentially a copy of []*models.HashicorpCloudPacker20230101Version, but without
+// the build and ancestry details
+type DatasourceOutput struct {
+	// Name of the author who created this version.
+	AuthorID string `mapstructure:"author_id"`
+
+	// The name of the bucket that this version is associated with.
+	BucketName string `mapstructure:"bucket_name"`
+
+	// Current state of the version.
+	Status string `mapstructure:"status"`
+
+	// The date the version was created.
+	CreatedAt string `mapstructure:"created_at"`
+
+	// The fingerprint of the version; this is a  unique identifier set by the Packer build
+	// that created this version.
+	Fingerprint string `mapstructure:"fingerprint"`
+
+	// The version ID. This is a ULID, which is a unique identifier similar
+	// to a UUID. It is created by the HCP Packer Registry when a version is
+	// first created, and is unique to this version.
+	ID string `mapstructure:"id"`
+
+	// The version name is created by the HCP Packer Registry once a version is
+	// "complete". Incomplete or failed versions currently default to having a name "v0".
+	Name string `mapstructure:"name"`
+
+	// The date when this version was last updated.
+	UpdatedAt string `mapstructure:"updated_at"`
+
+	// The ID of the channel used to query this version.
+	ChannelID string `mapstructure:"channel_id"`
+}
+
+func (d *Datasource) OutputSpec() hcldec.ObjectSpec {
+	return (&DatasourceOutput{}).FlatMapstructure().HCL2Spec()
+}
+
+func (d *Datasource) Execute() (cty.Value, error) {
+	ctx := context.TODO()
+
+	cli, err := hcpapi.NewClient()
+	if err != nil {
+		return cty.NullVal(cty.EmptyObject), err
+	}
+	log.Printf(
+		"[INFO] Reading HCP Packer Version info from HCP Packer Registry (%s) "+
+			"[project_id=%s, organization_id=%s, channel=%s]",
+		d.config.BucketName, cli.ProjectID, cli.OrganizationID, d.config.ChannelName,
+	)
+
+	channel, err := cli.GetChannel(ctx, d.config.BucketName, d.config.ChannelName)
+	if err != nil {
+		return cty.NullVal(cty.EmptyObject), fmt.Errorf(
+			"error retrieving HCP Packer Version from HCP Packer Registry: %s",
+			err.Error(),
+		)
+	}
+	if channel.Version == nil {
+		return cty.NullVal(cty.EmptyObject), fmt.Errorf(
+			"there is no HCP Packer Version associated with the channel %s",
+			d.config.ChannelName,
+		)
+	}
+
+	version := channel.Version
+
+	if version.Status == hcpPackerModels.HashicorpCloudPacker20230101VersionStatusVERSIONREVOKED.Pointer() {
```

Comment:
ditto on incorrect value check
```suggestion
	if *version.Status == hcpPackerModels.HashicorpCloudPacker20230101VersionStatusVERSIONREVOKED {
```